test/python/primitives/test_backend_estimator.py

```diff
@@ -254,6 +254,73 @@ def test_options(self, backend):
             self.assertIsInstance(result, EstimatorResult)
             np.testing.assert_allclose(result.values, [-1.307397243478641], rtol=0.1)
 
+    def test_job_size_limit_v2(self):
+        """Test BackendEstimator respects job size limit"""
+
+        class FakeNairobiLimitedCircuits(FakeNairobiV2):
+            """FakeNairobiV2 with job size limit."""
+
+            @property
+            def max_circuits(self):
+                return 1
+
+        backend = FakeNairobiLimitedCircuits()
+        backend.set_options(seed_simulator=123)
+        qc = QuantumCircuit(1)
+        qc2 = QuantumCircuit(1)
+        qc2.x(0)
+
+        op = SparsePauliOp.from_list([("I", 1)])
+        op2 = SparsePauliOp.from_list([("Z", 1)])
+
+        est = BackendEstimator(backend=backend)
+        result = est.run([qc], [op], [[]]).result()
+        self.assertIsInstance(result, EstimatorResult)
+        np.testing.assert_allclose(result.values, [1], rtol=0.1)
+
+        result = est.run([qc], [op2], [[]]).result()
+        self.assertIsInstance(result, EstimatorResult)
+        np.testing.assert_allclose(result.values, [1], rtol=0.1)
+
+        result = est.run([qc2], [op], [[]]).result()
+        self.assertIsInstance(result, EstimatorResult)
+        np.testing.assert_allclose(result.values, [1], rtol=0.1)
+
+        result = est.run([qc2], [op2], [[]]).result()
```

Comment:
Should we try to pass more than one circuit / observable at a time to the estimator and assert with `unittest.mock` that the `BackendV2.run` entry point only ever gets given one circuit at a time?

Reply:
ah, yeah good call. I just was lazy and copy and pasted the test from above and changed the backend but neglected to check we weren't passing >1 at a time. I'll fix this now